cmd/lncli/commands.go

```diff
@@ -728,28 +756,216 @@ func closeChannel(ctx *cli.Context) error {
 				return err
 			}
 
-			printJSON(struct {
-				ClosingTXID string `json:"closing_txid"`
-			}{
-				ClosingTXID: txid.String(),
-			})
+			txidChan <- txid.String()
 
 			if !ctx.Bool("block") {
 				return nil
 			}
-
 		case *lnrpc.CloseStatusUpdate_ChanClose:
-			closingHash := update.ChanClose.ClosingTxid
-			txid, err := chainhash.NewHash(closingHash)
+			return nil
+		}
+	}
+}
+
+var closeAllChannelsCommand = cli.Command{
+	Name:  "closeallchannels",
+	Usage: "Close all existing channels.",
+	Description: `
+	Close all existing channels.
+
+	Channels will be closed either cooperatively or unilaterally, depending
+	on whether the channel is active or not. If the channel is inactive, any
+	settled funds within it will be time locked for a few blocks before they
+	can be spent.
+
+	One can request to close inactive channels only by using the
+	--inactive_only flag.
+
+	By default, one is prompted for confirmation every time an inactive
+	channel is requested to be closed. To avoid this, one can set the
+	--force flag, which will only prompt for confirmation once for all
+	inactive channels and proceed to close them.`,
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "inactive_only",
+			Usage: "close inactive channels only",
+		},
+		cli.BoolFlag{
+			Name: "force",
+			Usage: "ask for confirmation once before attempting " +
+				"to close existing channels",
+		},
+	},
+	Action: actionDecorator(closeAllChannels),
+}
+
+func closeAllChannels(ctx *cli.Context) error {
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	listReq := &lnrpc.ListChannelsRequest{}
+	openChannels, err := client.ListChannels(context.Background(), listReq)
+	if err != nil {
+		return fmt.Errorf("unable to fetch open channels: %v", err)
+	}
+
+	if len(openChannels.Channels) == 0 {
+		return errors.New("no open channels to close")
+	}
+
+	channelsToClose := openChannels.Channels
+
+	switch {
+	case ctx.Bool("force") && ctx.Bool("inactive_only"):
+		msg := "Unilaterally close all inactive channels? The funds " +
+			"within these channels will be locked for some blocks " +
+			"(CSV delay) before they can be spent. (yes/no): "
+
+		confirmed := promptForConfirmation(msg)
+
+		// We can safely exit if the user did not confirm.
+		if !confirmed {
+			return nil
+		}
+
+		for i, channel := range channelsToClose {
+			// If the channel is active, remove it from our list
+			// as we were requested to close inactive channels only.
+			if channel.GetActive() {
+				channelsToClose = append(channelsToClose[:i],
```

Comment:
maaaybe it is easier to reason about this code if you start out with an empty slice and then *add* channels to `channelsToClose` instead of deleting them?

Reply:
Fixed.